assets/wizards/site-design/index.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * WordPress dependencies.
+ */
+import { Component, render, Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Material UI dependencies.
+ */
+import HeaderIcon from '@material-ui/icons/Web';
+
+/**
+ * Internal dependencies.
+ */
+import { withWizard } from '../../components/src';
+import Router from '../../components/src/proxied-imports/router'
+import { ThemeSelection } from './views';
+
+const { HashRouter, Redirect, Route, Switch } = Router;
+
+/**
+ * Site Design Wizard.
+ */
+class SiteDesignWizard extends Component {
+
+	componentDidMount = () => {
+		this.retrieveTheme();
+	};
+
+	retrieveTheme = () => {
+		const { setError, wizardApiFetch } = this.props;
+		const params = {
+			path: '/newspack/v1/wizard/newspack-setup-wizard/theme',
+			method: 'GET',
+		};
+		wizardApiFetch( params )
+			.then( response => {
+				this.setState( { theme: response.theme } );
+			} )
+			.catch( error => {
+				console.log( '[Theme Fetch Error]', error );
+				setError( { error } );
+			} );
+	};
+
+	updateTheme = theme => {
+		const { setError, wizardApiFetch } = this.props;
+		const params = {
+			path: '/newspack/v1/wizard/newspack-setup-wizard/theme/' + theme,
+			method: 'POST',
+		};
+		wizardApiFetch( params )
+			.then( response => {
+				this.setState( { theme: response.theme } );
+			} )
+			.catch( error => {
+				console.log( '[Theme Update Error]', error );
+				setError( { error } );
+			} );
+	};
+
+	state = {
+		theme: null,
+	};
+
+	/**
+	 * Render
+	 */
+	render() {
+		const { pluginRequirements } = this.props;
+		return (
+			<Fragment>
+				<HashRouter hashType="slash">
+					<Switch>
+						{ pluginRequirements }
+						<Route
+							path="/"
+							exact
+							render={ routeProps => {
+								const { theme } = this.state;
+								return (
+									<ThemeSelection
+										headerIcon={ <HeaderIcon /> }
+										headerText={ __( 'Site Design', 'newspack' ) }
+										subHeaderText={ __( 'Choose a Newspack theme', 'newspack' ) }
+										buttonText={ __( 'Customize', 'newspack' ) }
+										buttonAction='/wp-admin/customize.php?return=%2Fwp-admin%2Fadmin.php%3Fpage%3Dnewspack-site-design-wizard'
```

Comment:
I tried lopping off the `return` query param and found this still works. It appears WP automatically records the URL you came to Customizer from, with or without this param. Works either way, so 🤷‍♂

Reply:
No need to keep the long URL then 😄 
Removed in https://github.com/Automattic/newspack-plugin/pull/421/commits/cd53d229bdc0d1947cea11acc0da9f30bae12b4b